### PR TITLE
fix: snyk code enablement from preference dialog during authentication

### PR DESCRIPTION
**Pull request by bastiandoetsch**
### Description

The checkboxes remained disabled, as we asked the API without a token.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [x] CHANGELOG.md updated
- [ ] README.md updated, if user-facing

### Screenshots / GIFs

_Visuals that may help the reviewer. Please add screenshots for any UI change. GIFs are most welcome!_

### CODE REVIEW

File: src/main/kotlin/io/snyk/plugin/ui/settings/ScanTypesPanel.kt [hunk 1]
```diff
@@ -207,15 +207,20 @@ class ScanTypesPanel(
         }
     }
 
-    private fun checkSastEnable() {
+    fun checkSastEnabled(token: String? = pluginSettings().token) {
         setSnykCodeAvailability(false)
+        if (token.isNullOrBlank()) {
+            showSnykCodeAlert("A Snyk Token is necessary to check for Snyk Code enablement.")
```
Comment (michelkaporin) [inline src/main/kotlin/io/snyk/plugin/ui/settings/ScanTypesPanel.kt:213]: Where does this get shown? Screenshot would be great 🙏

File: src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt [hunk 2]
```diff
@@ -88,6 +92,13 @@ class SnykProjectSettingsConfigurable(val project: Project) : SearchableConfigur
             getSyncPublisher(project, SnykResultsFilteringListener.SNYK_FILTERING_TOPIC)?.filtersChanged()
             getSyncPublisher(project, SnykProductsOrSeverityListener.SNYK_ENABLEMENT_TOPIC)?.enablementChanged()
         }
+
+        runBackgroundableTask("Identifying with Analytics service", project, true) {
```
Comment (michelkaporin) [inline src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt:96]: Why was it moved here?

**Comment by michelkaporin** [comment]
Let's address new detekt warnings 🙏

File: src/main/kotlin/io/snyk/plugin/ui/settings/ScanTypesPanel.kt (hunk 1 above)
Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/ui/settings/ScanTypesPanel.kt:213]: It's only shown if no token is set, as we can only check Snyk Code enablement with a working API token.

File: src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt (hunk 2 above)
Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/settings/SnykProjectSettingsConfigurable.kt:96]: It was blocking the UI thread, which is bad - on my computer, Sentry & Amplitude are DNS blocked, so the connection times out only after 60s and would block the UI for that amount of time.

File: src/main/kotlin/io/snyk/plugin/services/SnykApiService.kt [hunk 3]
```diff
@@ -98,33 +97,37 @@
 
     private var currentUniqSnykApiClient: UniqSnykApiClient? = null
 
-    private fun getSnykApiClient(): SnykApiClient? {
+    private fun getSnykApiClient(token: String? = pluginSettings().token): SnykApiClient? {
```
[the remark this answers is not in the document]
Comment (bastiandoetsch) [reply src/main/kotlin/io/snyk/plugin/services/SnykApiService.kt:100]: Won't fix this. It is just a stupid detekt rule that actively discourages guard expressions. I'll deactivate / adjust it.

**Review by michelkaporin — APPROVED** [review]

**Pull request merged**